src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts

```diff
@@ -126,6 +128,19 @@ export class AcrossConfigStoreClient extends BaseAbstractClient {
     return across.rateModel.parseAndReturnRateModelFromString(config.routeRateModel[route]);
   }
 
+  /**
+   * Resolve the implied set of chain ID indices based on the chain ID of the ConfigStore.
+   * @param chainId Chain ID of the ConfigStore.
+   * @dev If the resolved chain ID is part of the default set, assume the protocol defaults.
+   *      Otherwise, assume this is a test deployment with a lone chain ID.
+   * @dev The protocol defaults are [1, 10, 137, 288, 42161] (outlined in UMIP-157).
+   * @dev chainId is marked optional to appease tsc. It must always be passed in.
+   */
+  protected implicitChainIdIndices(chainId?: number): number[] {
+    assert(isDefined(chainId), "ConfigStoreClient used before update");
+    return PROTOCOL_DEFAULT_CHAIN_ID_INDICES[0] === chainId ? PROTOCOL_DEFAULT_CHAIN_ID_INDICES : [chainId];
```

Comment:
I don't fully understand this logic, why should the first default chain ID index be [chainId]? Does that make sense? maybe it does

Reply:
As I see it, if we detect that we're on a deployment to anything other than chain ID 1, it means we're in some test/alt environment. In that case, all assumptions are thrown out the window and the ConfigStoreClient has no option but to assume it's the only chain ID in existence.

It might seem counter-intuitive to require that the caller to specify `chainId` as an input argument, rather than using `this.chainId`. The reason I did this is that we update the class state variables at the end of the `ConfigStoreClient.update()` method, but we need to call `implicitChainIdIndices()` in the middle of the `_update()` method. So in the case where we actually need to know the implicit chain IDs, `this.chainId` is actually undefined. So basically, it's done this way to be consistent with the rest of the implementation.